lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -361,19 +380,59 @@ static void replaceLoad(LoadInst *LI, SILValue val, AllocStackInst *ASI) {
     val = projection.createObjectProjection(builder, LI->getLoc(), val).get();
   }
   op = LI->getOperand();
-  LI->replaceAllUsesWith(val);
+  while (!LI->use_empty()) {
+    Operand *userOp = *LI->use_begin();
+
+    // We'll handle destorys later.
+    // Otherwise, make sure we copy and consume that.
+    if (LI->getOwnershipQualifier() == LoadOwnershipQualifier::Copy &&
+        userOp->isConsumingUse() && !isa<DestroyValueInst>(userOp->getUser()))
+      userOp->set(builder.createCopyValue(LI->getLoc(), val));
+    // If the user can't accept the projection, create a copy/destroy and hope
+    // that fixes it. Currently, the only time that this would happen is if
```

Comment:
Can you verify this is a correct statement:
> Currently, the only time that this would happen is if we're replacing a guaranteed object with an owned one.